oddt/spatial.py

```diff
@@ -150,20 +160,31 @@ def rmsd(ref, mol, ignore_h=True, method=None, normalize=False):
         ref_atoms = ref.atom_dict['coords'][np.hstack(ref_map)]
     elif method == 'min_symmetry':
         min_rmsd = None
-        ref_atoms = ref.coords[ref.atom_dict['atomicnum'] != 1]
-        mol_atoms = mol.coords[mol.atom_dict['atomicnum'] != 1]
-        for match in oddt.toolkit.Smarts(ref).findall(mol, unique=False):
-            match = np.array(match, dtype=int)
-            if oddt.toolkit.backend == 'ob':
-                match -= 1
-            rmsd = np.sqrt(((mol_atoms - ref_atoms[match])**2).sum(axis=-1).mean())
-            if min_rmsd is None or rmsd < min_rmsd:
-                min_rmsd = rmsd
-        return min_rmsd
+        ref_atoms = ref.atom_dict[ref.atom_dict['atomicnum'] != 1]['coords']
+        mol_atoms = mol.atom_dict[mol.atom_dict['atomicnum'] != 1]['coords']
+        # safety swith to check if number of heavy atoms match
+        if ref_atoms.shape == mol_atoms.shape:
+            # match mol to ref, generate all matches to find best RMSD
+            matches = oddt.toolkit.Smarts(ref).findall(mol, unique=False)
+            if not matches:
+                raise ValueError('Could not find any match between molecules.')
+            # calculate RMSD between all matches and retain the smallest
+            for match in matches:
+                match = np.array(match, dtype=int)
+                if is_openbabel_molecule(mol):
+                    match -= 1  # OB has 1-based indices
+                tmp_dict = mol.atom_dict[match]
+                mol_atoms = tmp_dict[tmp_dict['atomicnum'] != 1]['coords']
+                if mol_atoms.shape != ref_atoms.shape:
+                    raise ValueError('Molecular match got wrong number of '
+                                     'atoms.')
+                rmsd = np.sqrt(((mol_atoms - ref_atoms)**2).sum(axis=-1).mean())
+                if min_rmsd is None or rmsd < min_rmsd:
+                    min_rmsd = rmsd
+            return min_rmsd
     elif ignore_h:
-        hvy_map = np.argwhere(mol.atom_dict['atomicnum'] != 1).flatten()
-        mol_atoms = mol.coords[hvy_map]
-        ref_atoms = ref.coords[hvy_map]
+        mol_atoms = mol.coords[mol.atom_dict['atomicnum'] != 1]
+        ref_atoms = ref.coords[ref.atom_dict['atomicnum'] != 1]
     else:
         mol_atoms = mol.coords
```

Comment:
we don't test this variant

Reply:
This is too variable across different versions of OB/RDKit, so I didn't test it. What could go wrong there :)